src/model.ts

```diff
@@ -211,13 +210,20 @@ export class Model extends EventBus<any> implements CommandDispatcher {
     this.getters = {
       isReadonly: () => this.config.mode === "readonly" || this.config.mode === "dashboard",
       isDashboard: () => this.config.mode === "dashboard",
-      canUndo: this.history.canUndo.bind(this.history),
-      canRedo: this.history.canRedo.bind(this.history),
       getClient: this.session.getClient.bind(this.session),
       getConnectedClients: this.session.getConnectedClients.bind(this.session),
       isFullySynchronized: this.session.isFullySynchronized.bind(this.session),
     } as Getters;
 
+    this.history = new LocalHistory(
+      this.dispatchFromCorePlugin,
+      this.dispatch,
```

Comment:
I really feel dis-confortable with this. It drives us even more and more away from the plugin architecture. Going back will be harder and harder if we continue in this direction with funky/non-standard stuff.

If `this.session.on("revision-undone" ...` inside `LocalHistory` is moved to another (core) plugin (with session made available in the plugins config), we wouldn't need this double dispatch function and `LocalHistory` could be a "real" UI plugin ?

Reply:
Chanegd history into a real UI plugin, and handled the `resvision-redone/undone` inside the model.